src/main/scala/org/constellation/consensus/EdgeProcessor.scala

```diff
@@ -248,22 +271,80 @@ object EdgeProcessor extends StrictLogging {
     result.sequence
   }
 
-  // Temporary for testing join/leave logic.
+  def resolveTransactions(
+    hashes: Seq[String],
+    priorityPeer: Id
+  )(implicit dao: DAO): Future[Seq[TransactionCacheData]] = {
 
-  def handleSignatureRequest(sr: SignatureRequest)(implicit dao: DAO): SignatureResponse = {
-    //if (sr.facilitators.contains(dao.id)) {
-    // val replyTo = sr.checkpointBlock.witnessIds.head
-    val updated = if (sr.checkpointBlock.simpleValidation()) {
-      Some(hashSign(sr.checkpointBlock.baseHash, dao.keyPair))
-    } else {
-      None
+    implicit val exec: ExecutionContextExecutor = dao.signatureExecutionContext
+
+    def lookupTransaction(hash: String, client: APIClient): Future[Option[TransactionCacheData]] = {
+      async {
```

Comment:
I don't understand why did you wrapped this code with async/await as `getNonBlocking` is returning Future itself

Reply:
Interesting, I'm not entirely sure, these changes were merged in from the UI update/demo branch, will remove and retest.

Reply:
Ryle is going to update this on his pr to minimize merge headache.